lib/pbench/server/api/resources/api_key.py

```diff
@@ -64,23 +67,29 @@ def _get(
             Success: 200 with response containing the list of api_key
 
         Raises:
-            APIAbort, reporting "UNAUTHORIZED"
+            APIAbort, reporting "UNAUTHORIZED", "NOT_FOUND" and "FORBIDDEN"
         """
         user = Auth.token_auth.current_user()
+        key_id = params.uri.get("key")
 
         if not user:
             raise APIAbort(
                 HTTPStatus.UNAUTHORIZED,
                 "User provided access_token is invalid or expired",
             )
-
-        key = APIKey.query(user=user)
-        key_dict: JSONOBJECT = {}
-        if key:
-            for i in key:
-                key_dict[i.id] = {"name": i.name, "key": i.api_key}
-
-        response = {"api_key": key_dict}
+        if key_id:
+            key = APIKey.query(id=key_id)
```

Comment:
`key = APIKey.query(id=key_id, user=user)` will return a key only if there exists a key with the specified ID that's also owned by the authenticated user. That's why I like the `filter_by(**kwargs)` you wrote ... although I also think this should return `.all()` to be more general, it's not directly relevant *here* as this combination of filters will return one or none. (But `if not key:` will work exactly the same on `[]` as on `None`...)

(And as I mention elsewhere, security-wise it'd be slightly better *not* to distinguish here between "not found" and "not mine"; if it's "not mine" then, to me, it doesn't exist.)